sdk/src/client/api/block_builder/input_selection/requirement/ed25519.rs

```diff
@@ -27,7 +27,9 @@ impl InputSelection {
             .unwrap()
             .expect("expiration unlockable outputs already filtered out");
 
-        &required_address == address
+        required_address
+            .backing_ed25519()
+            .map_or(false, |a| &Address::Ed25519(*a) == address)
```

Comment:
I guess we could use `a == address.as_ed25519()` instead to avoid the copy. By now we should have the guarantee it's actually an ed25519.